modules/message-lane/src/instant_payments.rs

```diff
@@ -55,34 +65,252 @@ where
 		}
 	}
 
-	fn pay_relayer_reward(
-		_confirmation_relayer: &AccountId,
-		relayer: &AccountId,
-		reward: &Currency::Balance,
+	fn pay_relayers_rewards(
+		confirmation_relayer: &AccountId,
+		relayers_rewards: RelayersRewards<AccountId, Currency::Balance>,
 		relayer_fund_account: &AccountId,
 	) {
-		let pay_result = Currency::transfer(
-			&relayer_fund_account,
-			relayer,
-			*reward,
-			ExistenceRequirement::AllowDeath,
+		pay_relayers_rewards(
+			&mut CurrencyAdapter::<AccountId, Currency>(Default::default()),
+			confirmation_relayer,
+			relayers_rewards,
+			relayer_fund_account,
+			GetConfirmationFee::get(),
 		);
+	}
+}
+
+/// Shortened currency trait used from instant payments.
+trait InstantCurrency<AccountId, Balance> {
+	/// Transfer some free balance between accounts.
+	fn transfer(&mut self, source: &AccountId, dest: &AccountId, value: Balance) -> DispatchResult;
+}
+
+/// Currency adapter.
```

Comment:
It implements `InstantCurrency` for `Currency` (see my other comment about `InstantCurrency`)

Reply:
Can't we just blanket-impl `InstantCurrency` for any type that implements `Currency`? It seems like it's yet another abstraction and it's already quite complicated.

I do understand that the goal is to simplify writing tests, so that we don't have to set up the entire runtime, but at this point I'm really unsure if simplified test setup is worth this complexity. We already have mock runtime for message lane tests and it could simply just include balances module, which at least would make the test more end to end.

Reply:
@svyatonik could you comment on the above suggestions? Either blanket-impl or using actual mock runtime?

Reply:
https://github.com/paritytech/parity-bridges-common/pull/599/commits/c53794b7455af2c1d8efa3bf1814db9ac61fd458